app/components/goodcity/locations-search-overlay.js

```diff
@@ -43,7 +44,7 @@ export default SingletonComponent.extend(SearchMixin, {
 
     async selectLocation(location) {
       const callback = this.getWithDefault("onSelect", _.noop);
-
+      this.set("selectedLocation", location);
```

Comment:
We should **not** store a location in that component. It should only **return** it.
Whoever calls the component (via the service) can store the returned location and do whatever it needs with it.

This is a stateful red flag

Components should be designed in a generic manner, and should not bend to the desires of the controllers :)

Reply:
will check, and fix this @patrixr .